### PR TITLE
#269: Change Max and Min to a proper exception

### PR DESCRIPTION
**Pull request by fabriciofx**
As per #269 .

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256 please, pay attention to this pull request

**Comment by svendiedrichsen** [comment]
@fabriciofx: Sorry to annoy you again but if you call next() on an empty iterator the NoSuchElementException will be thrown automatically. You don't need to check for hasNext() and throw it yourself. Thus you can remove the if statement.

**Comment by fabriciofx** [comment]
@svendiedrichsen Don't apologize. I'm who should be thanking you, because it's one more opportunity to learn.  The problem is `next()` throws `NoSuchElementException` without a proper message. So, I think better in this case throws a `NoSuchElementException` with a message like "Can't find greater element in an empty interable". WDYT?

**Comment by svendiedrichsen** [comment]
I agree. Having a proper message is better.

File: src/main/java/org/cactoos/list/Max.java [hunk 1]
```diff
@@ -65,7 +65,9 @@ public Max(final Iterable<Scalar<T>> items) {
     public T value() throws Exception {
         final Iterator<Scalar<T>> iter = this.items.iterator();
         if (!iter.hasNext()) {
-            throw new IOException("Iterable is empty");
+            throw new NoSuchElementException(
+                "Can't find greater element in an empty interable"
```
Comment (alex-semenyuk) [inline src/main/java/org/cactoos/list/Max.java]: @fabriciofx Typo "interable"

Comment (fabriciofx) [inline src/main/java/org/cactoos/list/Max.java]: @alex-semenyuk I'll fix it! Thanks!

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**